utility.py

```diff
@@ -35,4 +36,24 @@ def distance(pos1: Tuple[float, float], pos2: Tuple[float, float]):
     """
     x1, y1 = pos1
     x2, y2 = pos2
-    return np.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)
\ No newline at end of file
+    return np.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)
+
+
+def firstn(n, gen):
+    """
+    Restricts generator to yields at most N element
+    """
+    for _ in range(0, n):
+        yield next(gen)
+
+
+def apply(func, gen):
+    """
+    Modifies each element in generator with a given function
+    """
+    if isinstance(gen, collections.Sequence):
+        # Convert the Sequence to a Generator
+        gen = iter(gen)
+
+    for x in gen:
+        yield func(x)
```

Comment:
This seems close to what the `map` function does. Doe `map` not support generators?

Reply:
Fixed in 2f5cbb2. I wasn't aware `map` existed, and searching the internet didn't yield anything for me